clevertap-core/src/main/java/com/clevertap/android/sdk/network/api/CtApi.kt

```diff
@@ -0,0 +1,112 @@
+package com.clevertap.android.sdk.network.api
+
+import android.net.Uri
+import com.clevertap.android.sdk.Logger
+import com.clevertap.android.sdk.network.http.CtHttpClient
+import com.clevertap.android.sdk.network.http.Request
+import com.clevertap.android.sdk.network.http.Response
+
+class CtApi(
+    private val httpClient: CtHttpClient,
+    val defaultDomain: String,
+    var domain: String?,
```

Comment:
why are these fields defined as var? do they change?

Reply:
`domain` and `spikyDomain` are changed from the corresponding methods of NetworkManager - `setDomain` and `setSpikyDomain`. `region`, `proxyDomain` and `spikyProxyDomain` are currently only used in tests, but from the perspective of the CtApi class I think we should still treat them as possibly changeable in order to be consistent with `domain` and `spikyDomain`.